src/dft/backends/backend_backward_instantiations.cxx

```diff
@@ -0,0 +1,65 @@
+#define PREC_F        dft::detail::precision::SINGLE
+#define PREC_D        dft::detail::precision::DOUBLE
+#define DESC_RF       dft::detail::descriptor<PREC_F, DOM_REAL>
+#define DESC_CF       dft::detail::descriptor<PREC_F, DOM_COMPLEX>
+#define DESC_RD       dft::detail::descriptor<PREC_D, DOM_REAL>
+#define DESC_CD       dft::detail::descriptor<PREC_D, DOM_COMPLEX>
+#define DEPENDS_VEC_T const std::vector<sycl::event> &
+
+#define ONEMKL_DFT_BACKWARD_INSTANTIATIONS(DESCRIPTOR_T, REAL_T, FORWARD_T, BACKWARD_T)       \
+    /* Buffer API */                                                                          \
+    template ONEMKL_EXPORT void compute_backward<DESCRIPTOR_T, BACKWARD_T>(                   \
+        DESCRIPTOR_T & desc, sycl::buffer<BACKWARD_T> &);                                     \
+    template ONEMKL_EXPORT void compute_backward<DESCRIPTOR_T, BACKWARD_T, FORWARD_T>(        \
+        DESCRIPTOR_T & desc, sycl::buffer<BACKWARD_T> &, sycl::buffer<FORWARD_T> &);          \
+    template ONEMKL_EXPORT void compute_backward<DESCRIPTOR_T, REAL_T>(                       \
+        DESCRIPTOR_T & desc, sycl::buffer<REAL_T> &, sycl::buffer<REAL_T> &);                 \
+    template ONEMKL_EXPORT void compute_backward<DESCRIPTOR_T, REAL_T, REAL_T>(               \
+        DESCRIPTOR_T & desc, sycl::buffer<REAL_T> &, sycl::buffer<REAL_T> &,                  \
+        sycl::buffer<REAL_T> &, sycl::buffer<REAL_T> &);                                      \
+                                                                                              \
+    /* USM API */                                                                             \
+    template ONEMKL_EXPORT sycl::event compute_backward<DESCRIPTOR_T, BACKWARD_T>(            \
+        DESCRIPTOR_T & desc, BACKWARD_T *, DEPENDS_VEC_T);                                    \
+    template ONEMKL_EXPORT sycl::event compute_backward<DESCRIPTOR_T, BACKWARD_T, FORWARD_T>( \
+        DESCRIPTOR_T & desc, BACKWARD_T *, FORWARD_T *, DEPENDS_VEC_T);                       \
+    template ONEMKL_EXPORT sycl::event compute_backward<DESCRIPTOR_T, REAL_T>(                \
+        DESCRIPTOR_T & desc, REAL_T *, REAL_T *, DEPENDS_VEC_T);                              \
+    template ONEMKL_EXPORT sycl::event compute_backward<DESCRIPTOR_T, REAL_T, REAL_T>(        \
+        DESCRIPTOR_T & desc, REAL_T *, REAL_T *, REAL_T *, REAL_T *, DEPENDS_VEC_T);
+
+ONEMKL_DFT_BACKWARD_INSTANTIATIONS(DESC_RF, float, float, std::complex<float>)
+ONEMKL_DFT_BACKWARD_INSTANTIATIONS(DESC_CF, float, std::complex<float>, std::complex<float>)
+ONEMKL_DFT_BACKWARD_INSTANTIATIONS(DESC_RD, double, double, std::complex<double>)
+ONEMKL_DFT_BACKWARD_INSTANTIATIONS(DESC_CD, double, std::complex<double>, std::complex<double>)
+
+#undef DOM_REAL
+#undef DOM_COMPLEX
+#undef PREC_F32
+#undef PREC_F64
+#undef DESC_RF32
+#undef DESC_CF32
+#undef DESC_RF64
+#undef DESC_CF64
+#undef DEPENDS_VEC_T
```

Comment:
Shouldn't we also undef `ONEMKL_DFT_BACKWARD_INSTANTIATIONS`? Same for `FORWARD`.

Reply:
Done in [82edd19](https://github.com/hjabird/oneMKL/pull/4/commits/82edd192882ce9bd71c41fe57923ee93eb281f09).